neuralprophet/forecaster.py

```diff
@@ -2265,14 +2265,20 @@ def _make_future_dataframe(self, df, events_df, regressors_df, periods, n_histor
         if (n_historic_predictions + self.max_lags) == 0:
             df = pd.DataFrame(columns=df.columns)
         else:
+            # FIX ISSUE #498
+            last_valid_index = df["y"].last_valid_index()
+            last_index = df["y"].index[-1]
+            last_non_nan_index = last_index - last_valid_index + 1
             df = df[-(self.max_lags + n_historic_predictions) :]
-            if np.isnan(df["y"]).any():
-                raise ValueError(
-                    "Data used for historic forecasts contains NaN values. "
-                    "Please ensure there are no NaN values within the last {} entries of the df".format(
-                        self.max_lags + n_historic_predictions
-                    )
-                )
+            nan_at_end = 0
+            while len(df) > nan_at_end and df["y"].isnull().iloc[-(1 + nan_at_end)]:
+                nan_at_end += 1
+            fill_thresh = nan_at_end - n_historic_predictions
+            if fill_thresh > 0:
```

Comment:
Why only if there are more nans than historic prections?
Should we not always do this? (except if there is some excessive amount of missing values - and then we should trow a warning or error)